internal/gomod/module.go

```diff
@@ -60,7 +60,7 @@ func ListModules(projectRoot string) (string, map[string]string, error) {
 		return "", nil, nil
 	}
 
-	out, err := run(projectRoot, "go", "list", "-m", "all")
+	out, err := run(projectRoot, "go", "list", "-mod=readonly", "-m", "all")
```

Comment:
Why this addition? Will this require that mod already be fetched?

Reply:
Without this, a package with vendored dependencies can't be listed. It should be harmless for any package without vendored dependencies. It does require that the dependencies be already loaded, but if that isn't the case then the indexer mutates the module cache which seems wrong.

We could add a `go mod download` step first, though.

Reply:
To the tool or to the CI step?

Reply:
Either or both.

Reply:
For further context: I think it was a mistake for us to write the indexer in such a way that it will touch the `go` tool for anything other than reading metadata. Indexing should either be done _during_ a build, so that the dependencies for each package being indexed are already populated, or _after_ a build so that the dependencies of the whole module are populated.

But that ship may have already sailed, in which case doing it here may be the least-pessimal workaround.

Reply:
I added a call to `go mod download`. That preserves the existing behaviour for modules without vendored dependencies.